GenderPayGap.UnitTests/GenderPayGap.WebUI.Tests/Views/ManageOrganisations/ManageOrganisation.cs

```diff
@@ -0,0 +1,89 @@
+﻿using System;
+using GenderPayGap.Core;
+using GenderPayGap.Database;
+using GenderPayGap.Database.Models;
+using GenderPayGap.Extensions;
+using GenderPayGap.Tests.Common.TestHelpers;
+using GenderPayGap.WebUI.Models.ManageOrganisations;
+using GenderPayGap.WebUI.Tests.TestHelpers;
+using NUnit.Framework;
+
+namespace GenderPayGap.WebUI.Tests.Views.ManageOrganisations
+{
+    [TestFixture]
+    public class ManageOrganisation
+    {
+        private static readonly int ReportingYear = DateTime.Now.Year;
+        private const int PastReportingYear = 2018;
+        private readonly User user = UserHelpers.CreateUsers().Find(u => u.UserId == 24572);
+        
+        private readonly Organisation organisationOutOfScope = OrganisationHelper.GetOrganisationInAGivenScope(ScopeStatuses.OutOfScope, null, ReportingYear);
+        private readonly Organisation organisationInScope = OrganisationHelper.GetOrganisationInAGivenScope(ScopeStatuses.InScope, null, ReportingYear);
+        private readonly Organisation organisationInScopeForPastYear = OrganisationHelper.GetOrganisationInAGivenScope(ScopeStatuses.InScope, null, PastReportingYear);
+        
+        
+        [SetUp]
+        public void Setup()
+        {
+            VirtualDateTime.Initialise(TimeSpan.Zero);
+        }
+        
+        [Test]
+        public void WhenOnManageOrganisationPage_IfOrganisationIsNotRequiredToReport_ThenReportStatusIsNotRequired()
+        {
+            var viewModel = new ManageOrganisationDetailsForYearViewModel(organisationOutOfScope, ReportingYear, new DraftReturn());
+
+            string reportStatusText = viewModel.GetReportStatusText();
+            
+            Assert.AreEqual("Report not required", reportStatusText);
+        }
+        
+        [Test]
+        public void WhenOnManageOrganisationPage_IfHasNotSubmittedBeforeTheDeadline_ThenReportStatusIsDueByDate()
+        {
+            var viewModel = new ManageOrganisationDetailsForYearViewModel(organisationInScope, ReportingYear, null);
+
+            string reportStatusText = viewModel.GetReportStatusText();
+            
+            Assert.True(reportStatusText.Contains("Report due by"));
```

Comment:
Can't you check the deadline as well?

Reply:
I initially tried hard coding a year that I was sure wouldn't have any change in how the reporting deadline logic behaved i.e. 2021. It then occurred to me that hardcoding a year would mean the some of the tests will eventually fail as the reports would eventually become overdue

I then tried asserting `"Report due by 4 April" + ReportingYear+1`, I realised this wouldn't work either because the deadline would either be on the `ReportingYear` or on the `ReportingYear+1` depending on what time of the year we're on. Adding a check for this would have been more complicated than the test needed to be.

I eventually settled on this version because I felt the test was just meant to be checking what _kind_ of tag was being returned and not necessarily if the dates are accurate as another part of the code should be testing that logic. And this method meant the test wouldn't fail depending on the time of the year we are in

Reply:
Makes sense, thanks